docker/production/docker-compose.yml

```diff
@@ -0,0 +1,12 @@
+version: "3"
+
+services:
+  server:
+    build:
+      context: ./../..
+      dockerfile: ./docker/production/Dockerfile
+    image: unsynced/compass:latest # TODO(amantri): Set this to a shared account
+    platform: linux/amd64
+    command: bash -l -c "npm run start"
+    ports:
+      - "3000:3000"
```

Comment:
We discussed it, and I suggested keeping it then, so as to be similar to other docker configs in our project. `docker compose build && docker compose up` works for this, assuming you have a .env.production file in the project root dir (it can be a copy of .env.local) that specifies the needed environment flags. Added a comment for environment.

Reply:
sorry, I might be confused

when would you want to do `docker compose up` for this stack (in its current state)?

Reply:
To test it locally. I used that to debug issues on the image. TBF, `docker compase up` is nice to have.

`docker compose build` is easier to work with than `docker build` - I tried (not very hard) to get `docker build` to work, but minor things like context etc. were not being correctly set, so I gave up on that and stuck with `docker compose`. I can try harder to get `docker build` to work :)

Reply:
I agree it's useful to have for debugging images, but in that case wouldn't you want a way of bringing up the rest of the stack (Postgres) as well?

forgot to mention it earlier but it's convention to have `Dockerfile` at the root unless impossible which would fix the context issue when building

also not sure if you saw yet but I removed the docker dev workflow in my PR: https://github.com/sfbrigade/compass/pull/85

Reply:
Ah ok - just saw the changes on #85. Removing option 1 is fine, but we do need to keep the docker configs for prod and dev, so it might make sense to keep the docker directory.

Re Postgres and other services in the prod image: specifying DATABASE_URL and related vars in .env.production will allow you to run the image (i.e. option 3). I didn't include postgres in the prod version as it is not really needed for the actual production environment.

Reply:
> I didn't include postgres in the prod version as it is not really needed for the actual production environment.

Maybe we're talking about different things, I was advocating for having a stack that matches production services, not one that connects to them.

> Removing option 1 is fine, but we do need to keep the docker configs for prod and dev, so it might make sense to keep the docker directory.

At that point we'll only have 1 Dockerfile, so I would push for it being at the repo root to match convention. I can do that in my PR instead if this gets merged first since there's still 2 Dockerfiles in this PR context.

Reply:
> Maybe we're talking about different things, I was advocating for having a stack that matches production services, not one that connects to them.

I think we are talking about the same thing. Just to be sure, I am not suggesting that an engineer connect to prod services from a local server - in fact, the networking settings won't allow that. To run the production image, you would need to bring up postgres and other services locally (old option 2 or 3) and connect to them via env vars. This is basically the same as the dev stack where we run the Compass server locally, and connect to postgres, but instead of running the dev binary, you will be running the prod binary. We can share the docker-compose.yml for the supporting files.


> At that point we'll only have 1 Dockerfile, so I would push for it being at the repo root to match convention. I can do that in my PR instead if this gets merged first since there's still 2 Dockerfiles in this PR context.

Yes, but it might still be better to differentiate them via some folder structure as the Dockerfile will be for production Compass binary, while the docker-compose.yml will be for the supporting services. Putting them in the same directory suggests that they are related, while in reality they aren't. Maybe the current "docker/postgres" folder should be renamed to "docker/3p_services" or similar to indicate the supporting services.

Reply:
> This is basically the same as the dev stack where we run the Compass server locally, and connect to postgres, but instead of running the dev binary, you will be running the prod binary. We can share the docker-compose.yml for the supporting services.

What's the difference between the dev and prod service version in this context? Wouldn't we just have one prod Dockerfile?

Reply:
Debug build vs optimized prod build of our app (Compass)

Reply:
Updated the PR per our discussion yesterday. Changes:
 - Running db:migration in CMD (now using shell form instead of exec form as there are multiple commands)
 - moved docker-compose.yml to supporting_services/